includes/discovery/vlans/vlan_functions.inc.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/* idea from http://php.net/manual/en/function.hex2bin.php comments */
+function hex2bin_compat($str)
+{
+    if (strlen($str) % 2 !== 0) {
+        trigger_error(__FUNCTION__.'(): Hexadecimal input string must have an even length', E_USER_WARNING);
```

Comment:
This is what the actual hex2bin() function from php > 5.4 does, AFAIK.  Just trying to mimic its behavior.